toolchain/sem_ir/facet_type_info.cpp

```diff
@@ -42,21 +51,21 @@ static auto RequiredLess(const IdentifiedFacetType::RequiredInterface& lhs,
 
 // Assuming both `a` and `b` are sorted and deduplicated, replaces `a` with `a -
 // b` as sets. Assumes there are few elements between them.
-static auto SubtractSorted(
-    llvm::SmallVector<FacetTypeInfo::ImplsConstraint>& a,
-    const llvm::SmallVector<FacetTypeInfo::ImplsConstraint>& b) -> void {
-  using Iter = llvm::SmallVector<FacetTypeInfo::ImplsConstraint>::iterator;
+template <typename ConstraintT, typename CompareT>
+static auto SubtractSorted(llvm::SmallVector<ConstraintT>& a,
+                           const llvm::SmallVector<ConstraintT>& b,
+                           CompareT compare) -> void {
```

Comment:
Could `CompareT` be a `function_ref` instead of a template?

Reply:
For me, being able to see types on function boundaries seems helpful. But also, note there's a bit of a difference between `SortAndDeduplicate` where you can see immediately how it's used because it's a 2 line body, versus `SubtractSorted`.

If you want to make it look better, you might consider something like:

```
template<typename ConstRefT>
using LessThanFn = llvm::function_ref<auto(ConstRefT, ConstRefT)->bool>;
```

Particularly since you have the same function as parameter to two functions.

I'd also note, previously you were templating on `ConstraintT`, it's not obvious to me why you switched to `VecT`. But it seems like that switch is also part of why you're writing `typename Vect::const_reference` instead of `ConstraintT`.